packages/asset-service/src/generateAssetData/generateAssetData.ts

```diff
@@ -58,19 +80,24 @@ const generateAssetData = async () => {
               return { ...uniqueTokens[key], icon }
             }
           })
+
           modifiedTokens = modifiedTokens.concat(newModifiedTokens)
         }
+
         const baseAssetWithTokens: BaseAsset = {
           ...baseAsset,
           tokens: modifiedTokens
         }
+
         return baseAssetWithTokens
       } else {
         return baseAsset
       }
     })
   )
 
+  generatedAssetData = filterBlacklistedAssets(generatedAssetData)
```

Comment:
one last requested change, please assign to a new variable rather than mutating here. `filterBlacklistedAssets` should be pure, take all the assets and return the filtered subset, and not mutate it's inputs.

Reply:
You're right!